server/issue_parser.go

```diff
@@ -0,0 +1,62 @@
+// See License for license information.
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	jira "github.com/andygrunwald/go-jira"
+
+	"github.com/mattermost/mattermost-server/model"
+)
+
+var jiraLinkWithTextRegex = regexp.MustCompile(`\[([^\[]+)\|([^\]]+)\]`)
+
+func parseJiraLinksToMarkdown(text string) string {
+	return jiraLinkWithTextRegex.ReplaceAllString(text, "[${1}](${2})")
+}
+
+func mdKeySummaryLink(issue *jira.Issue) string {
+	pos := strings.LastIndex(issue.Self, "/rest/api")
+	if pos < 0 {
+		return ""
+	}
+	return fmt.Sprintf("[%s](%s%s)", issue.Key+": "+issue.Fields.Summary, issue.Self[:pos], "/browse/"+issue.Key)
+}
+
+func parseIssue(issue *jira.Issue) []*model.SlackAttachment {
+	text := mdKeySummaryLink(issue)
+	fmt.Printf("%q", issue.Fields.Description)
+	desc := truncate(issue.Fields.Description, 3000)
+	desc = parseJiraLinksToMarkdown(desc)
+	if desc != "" {
+		text += "\n\n" + desc + "\n"
+	}
+
+	var fields []*model.SlackAttachmentField
+	if issue.Fields.Assignee != nil {
+		fields = append(fields, &model.SlackAttachmentField{
+			Title: "Assignee",
+			Value: issue.Fields.Assignee.DisplayName,
+			Short: true,
+		})
+	}
+	if issue.Fields.Priority != nil {
+		fields = append(fields, &model.SlackAttachmentField{
+			Title: "Priority",
+			Value: issue.Fields.Priority.Name,
+			Short: true,
+		})
+	}
+	return []*model.SlackAttachment{
+		{
+			// TODO is this supposed to be themed?
+			Color:  "#95b7d0",
```

Comment:
I don't have the answer for this TODO, but I assume this hardcoded value can be pulled from somewhere.

Reply:
I bet, but I don't know where, and the original webhook code also has it hardcoded. I think it's probably a different PR -- to pull in the client's theme somehow.